### PR TITLE
firstSlider

### PR DESCRIPTION
**Pull request by asmd564**
https://fe-feb23-code-hunters.github.io/fe_phone-catalog/

### CODE REVIEW

File: src/components/ui/sliders/mainSlider/swiper/navigation.scss [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    cursor: auto;
+    pointer-events: none;
+  }
+  &.swiper-button-hidden {
+    opacity: 0;
+    cursor: auto;
+    pointer-events: none;
+  }
+  .swiper-navigation-disabled & {
+    display: none !important;
```
Comment (prokivskaaa) [inline src/components/ui/sliders/mainSlider/swiper/navigation.scss:27]: try not to use important

File: src/components/ui/sliders/mainSlider/swiper/navigation.scss [hunk 2]
```diff
@@ -0,0 +1,57 @@
+.swiper-button-prev,
+.swiper-button-next {
+  position: absolute;
+  top: 0;
+  left: 0;
+  right: 0;
+  max-height: 400px;
+  height: 100%;
+  width: 32px;
+  border: 1px solid #b4bdc4;
```
Comment (prokivskaaa) [inline src/components/ui/sliders/mainSlider/swiper/navigation.scss:10]: colors from vars

File: src/components/ui/sliders/mainSlider/swiper/navigation.scss (hunk 1 above)
Comment (asmd564) [reply src/components/ui/sliders/mainSlider/swiper/navigation.scss:27]: there are native styles of swiper js

File: src/components/ui/sliders/mainSlider/swiper/navigation.scss [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  &.swiper-button-hidden {
+    opacity: 0;
+    cursor: auto;
+    pointer-events: none;
+  }
+  .swiper-navigation-disabled & {
+    display: none !important;
+  }
+  &:after {
+    font-family: swiper-icons;
```
Comment (prokivskaaa) [inline src/components/ui/sliders/mainSlider/swiper/navigation.scss:30]: use already-made typography

File: src/components/ui/sliders/mainSlider/swiper/navigation.scss [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    color: #b4bdc4;
+    font-size: var(--swiper-navigation-size);
+    text-transform: none !important;
```
Comment (prokivskaaa) [inline src/components/ui/sliders/mainSlider/swiper/navigation.scss:33]: no important

File: src/components/ui/sliders/mainSlider/swiper/navigation.scss [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    cursor: auto;
+    pointer-events: none;
+  }
+  .swiper-navigation-disabled & {
+    display: none !important;
+  }
+  &:after {
+    font-family: swiper-icons;
+    color: #b4bdc4;
+    font-size: var(--swiper-navigation-size);
```
Comment (prokivskaaa) [inline src/components/ui/sliders/mainSlider/swiper/navigation.scss:32]: 1. use scss vars
2. there is already-made typography

File: src/components/ui/sliders/mainSlider/swiper/navigation.scss [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    font-variant: initial;
+    line-height: 1;
+  }
+}
+.swiper-button-prev,
+.swiper-rtl .swiper-button-next {
+  &:after {
+    content: 'prev';
+  }
+  left: var(--swiper-navigation-sides-offset, 10px);
```
Comment (prokivskaaa) [inline src/components/ui/sliders/mainSlider/swiper/navigation.scss:44]: scss vars

File: src/components/ui/sliders/mainSlider/swiper/navigation.scss [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+  }
+  .swiper-navigation-disabled & {
+    display: none !important;
+  }
+  &:after {
+    font-family: swiper-icons;
+    color: #b4bdc4;
+    font-size: var(--swiper-navigation-size);
+    text-transform: none !important;
+    letter-spacing: 0;
```
Comment (prokivskaaa) [inline src/components/ui/sliders/mainSlider/swiper/navigation.scss:34]: useless

File: src/components/ui/sliders/mainSlider/swiper/navigation.scss [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+  }
+  left: var(--swiper-navigation-sides-offset, 10px);
+  right: auto;
+}
+.swiper-button-next,
+.swiper-rtl .swiper-button-prev {
+  &:after {
+    content: 'next';
+  }
+  right: var(--swiper-navigation-sides-offset, 10px);
```
Comment (prokivskaaa) [inline src/components/ui/sliders/mainSlider/swiper/navigation.scss:52]: scss vars

File: src/components/ui/sliders/mainSlider/swiper/pagination.scss [hunk 9]
```diff
@@ -0,0 +1,160 @@
+.swiper-pagination {
+  position: absolute;
+  text-align: center;
+  transition: 300ms opacity;
```
Comment (prokivskaaa) [inline src/components/ui/sliders/mainSlider/swiper/pagination.scss:4]: 0.3s

File: src/components/ui/sliders/mainSlider/swiper/pagination.scss [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+  transform: translate3d(0, 0, 0);
+  z-index: 10;
+  &.swiper-pagination-hidden {
+    opacity: 0;
+  }
+  .swiper-pagination-disabled > &,
+  &.swiper-pagination-disabled {
+    display: none !important;
```
Comment (prokivskaaa) [inline src/components/ui/sliders/mainSlider/swiper/pagination.scss:12]: no important

File: src/components/ui/sliders/mainSlider/swiper/pagination.scss [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+  }
+}
+/* Common Styles */
```
Comment (prokivskaaa) [inline src/components/ui/sliders/mainSlider/swiper/pagination.scss:15]: delete comments

File: src/components/ui/sliders/mainSlider/swiper/pagination.scss [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+  }
+  .swiper-pagination-bullet-active-next {
+    transform: scale(0.66);
+  }
+  .swiper-pagination-bullet-active-next-next {
+    transform: scale(0.33);
+  }
+}
+.swiper-pagination-bullet {
+  width: var(--swiper-pagination-bullet-width, var(--swiper-pagination-bullet-size, 8px));
```
Comment (prokivskaaa) [inline src/components/ui/sliders/mainSlider/swiper/pagination.scss:53]: scss vars

File: src/components/ui/sliders/mainSlider/swiper/pagination.scss [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+    padding: 0;
+    box-shadow: none;
+    appearance: none;
+  }
+  .swiper-pagination-clickable & {
+    cursor: pointer;
+  }
+
+  &:only-child {
+    display: none !important;
```
Comment (prokivskaaa) [inline src/components/ui/sliders/mainSlider/swiper/pagination.scss:71]: no important

File: src/components/ui/sliders/mainSlider/swiper/pagination.scss [hunk 14]
```diff
@@ -0,0 +1,160 @@
...
+  }
+}
+.swiper-pagination-bullet-active {
+  opacity: var(--swiper-pagination-bullet-opacity, 1);
+  background: #313237;
```
Comment (prokivskaaa) [inline src/components/ui/sliders/mainSlider/swiper/pagination.scss:76]: use already-made vars

File: src/components/ui/sliders/mainSlider/swiper/pagination.scss [hunk 15]
```diff
@@ -0,0 +1,160 @@
...
+}
+
+.swiper-vertical > .swiper-pagination-bullets,
+.swiper-pagination-vertical.swiper-pagination-bullets {
+  right: var(--swiper-pagination-right, 8px);
+  left: var(--swiper-pagination-left, auto);
+  top: 50%;
+  transform: translate3d(0px, -50%, 0);
+  .swiper-pagination-bullet {
+    margin: var(--swiper-pagination-bullet-vertical-gap, 6px) 0;
```
Comment (prokivskaaa) [inline src/components/ui/sliders/mainSlider/swiper/pagination.scss:86]: scss vars

File: src/components/ui/sliders/mainSlider/swiper/pagination.scss [hunk 16]
```diff
@@ -0,0 +1,160 @@
...
+    .swiper-pagination-bullet {
+      display: inline-block;
+      transition: 200ms transform, 200ms top;
+    }
+  }
+}
+.swiper-horizontal > .swiper-pagination-bullets,
+.swiper-pagination-horizontal.swiper-pagination-bullets {
+  .swiper-pagination-bullet {
+    margin: 0 var(--swiper-pagination-bullet-horizontal-gap, 4px);
```
Comment (prokivskaaa) [inline src/components/ui/sliders/mainSlider/swiper/pagination.scss:102]: scss vars

File: src/components/ui/sliders/mainSlider/swiper/pagination.scss [hunk 17]
```diff
@@ -0,0 +1,160 @@
...
+    }
+  }
+}
+.swiper-horizontal.swiper-rtl > .swiper-pagination-bullets-dynamic .swiper-pagination-bullet,
+:host(.swiper-horizontal.swiper-rtl) .swiper-pagination-bullets-dynamic .swiper-pagination-bullet {
+  transition: 200ms transform, 200ms right;
+}
+/* Fraction */
+.swiper-pagination-fraction {
+  color: var(--swiper-pagination-fraction-color, inherit);
```
Comment (prokivskaaa) [inline src/components/ui/sliders/mainSlider/swiper/pagination.scss:119]: scss vars

File: src/components/ui/sliders/mainSlider/swiper/pagination.scss [hunk 18]
```diff
@@ -0,0 +1,160 @@
...
+}
+/* Progress */
+.swiper-pagination-progressbar {
+  background: var(--swiper-pagination-progressbar-bg-color, rgba(0, 0, 0, 0.25));
```
Comment (prokivskaaa) [inline src/components/ui/sliders/mainSlider/swiper/pagination.scss:123]: scss vars

File: src/components/ui/sliders/mainSlider/swiper/pagination.scss [hunk 19]
```diff
@@ -0,0 +1,160 @@
...
+  position: absolute;
+  .swiper-pagination-progressbar-fill {
+    background: var(--swiper-pagination-color, var(--swiper-theme-color));
```
Comment (prokivskaaa) [inline src/components/ui/sliders/mainSlider/swiper/pagination.scss:126]: scss vars

File: src/components/ui/sliders/mainSlider/swiper/pagination.scss [hunk 20]
```diff
@@ -0,0 +1,160 @@
...
+    width: 100%;
+    height: var(--swiper-pagination-progressbar-size, 4px);
+    left: 0;
+    top: 0;
+  }
+  .swiper-vertical > &,
+  &.swiper-pagination-vertical,
+  .swiper-horizontal > &.swiper-pagination-progressbar-opposite,
+  &.swiper-pagination-horizontal.swiper-pagination-progressbar-opposite {
+    width: var(--swiper-pagination-progressbar-size, 4px);
```
Comment (prokivskaaa) [inline src/components/ui/sliders/mainSlider/swiper/pagination.scss:152]: scss vars

File: src/components/ui/sliders/productSlider/productSlider.module.scss [hunk 21]
```diff
@@ -0,0 +1,8 @@
+@import '../../../../styles/fonts/fonts.scss';
```
Comment (prokivskaaa) [inline src/components/ui/sliders/productSlider/productSlider.module.scss:1]: already imported globally

File: src/components/ui/sliders/productSlider/swiper/navigation.scss [hunk 22]
```diff
@@ -0,0 +1,57 @@
+.swiper-button-prev,
+.swiper-button-next {
+  position: absolute;
+  top: 0;
+  left: 0;
+  right: 0;
+  max-height: 400px;
+  height: 100%;
+  width: 32px;
+  border: 1px solid #b4bdc4;
```
Comment (prokivskaaa) [inline src/components/ui/sliders/productSlider/swiper/navigation.scss:10]: the same situation with the:
1. scss vars
2. comments
4. already-made typography and vars

Comment (prokivskaaa) [reply src/components/ui/sliders/productSlider/swiper/navigation.scss:10]: 4. important

**Review by prokivskaaa — CHANGES_REQUESTED** [review]
Besides those comments from the review, try to improve styling in general by grouping info and inheritance (use mixins and placeholders). Delete redundant classes and styling, which do not contribute to the final result. Try to place styles in order, here is the tip: https://9elements.com/css-rule-order/ 

Also, make the buttons not transparent. Maybe use buttons from already-made components, but add some additional styling.

I understand that you took this code as a base for a final realization of Swiper, but please take care of styling to not make any confusion in the future.

**Pull request closed without merging**